### ISSUE TITLE
Value properties and tagged values are not being sent/saved on the server

### ISSUE THREAD

**Issue by andyko** [Open-MBEE/MDK#8]
I installed OpenMBEE on a server using Alfresco 4.2f. I tested the server using MagicDraw 14.0.5 with MDK installed. My test project initialized, and I managed to get most of my elements sent and saved on the server. I could even get the Document Views to show up. I did run into a particular problem : None of the Default Values of my value properties or my tagged values are being saved on the server. The value property itself is saved (i.e. I can see the value property in DocWeb), but I can't get it's default value to the server. Commiting the value (right click on the error in the Validation window) results in a message that it is "Sending ..." then "Successful." However, when I validate the model, the same error "[VALUE] model: not null, web: null" . The same thing is true for tagged values.

Any ideas why this is happening? Perhaps point me to a log file either on the server or on the MagicDraw side that will shed some light or clues as to what's happening?

Andy Ko

**Comment by dlamoris** [comment]
You mean magicdraw 17.0.5 right...?
What kind of values are they? If they're element references, it could be the referenced elements were not found on the server.

**Comment by dlamoris** [comment]
there should also be a log what got sent and received in the magicdraw log (md.log or magicdraw.log)

**Comment by tomasjkn** [comment]
> the same error "[VALUE] model: not null, web: null" 
Sounds eerily familiar :). This is the same stuff I have been fighting 2 months ago...
>You mean magicdraw 17.0.5 right...?
Probably yes - there is no such thing as MD v14.0.5. Only MDv17.0.x series had a 3-tier numbering scheme (due to some questionable upper management decisions :^) )
>What kind of values are they? If they're element references, ...
As far as I remember, it fails on simple values in the model - strings , reals, booleans. But maybe Andy has a different case.
>>Any ideas why this is happening? Perhaps point me to a log file either on the server 
>> or on the MagicDraw side that will shed some light or clues as to what's happening?
>there should also be a log what got sent and received in the magicdraw log (md.log or magicdraw.log)
Client side - MagicDraw log  - is under <your user home>/AppData/Local/.magicdraw/magicdraw.log (on Windows) or <your user home>/.magicdraw/magicdraw.log  (on Unixes).
There should be a fragment of JSON somewhere in there - the data that is being sent to server.

Sincerely,
Tomas J.

**Comment by andyko** [comment]
@dlamoris Yes - it's MagicDraw 17.0.5 - my mistake. I updated the posting to reflect this.

**Comment by andyko** [comment]
@dlamoris I'm seeing that the Default Value of value properties aren't being saved on the server. Here's an example - I have a block (let's call it 'test') in a package. In that block, I have a SysML Value Property called 'foo'. The Default Value of 'foo' is set to '20.5'. When I commit to OpenMBEE, the block 'test', and the value property 'foo' gets saved to the OpenMBEE server. To test, I can open DocWeb, and edit a View. In the edit dialog, I can click on "cf" and look for 'foo'. 'foo' shows up but no value is assigned to it. When I validate my model in MagicDraw, I get the "[VALUE] model: not null, web: null" error. When I commit the value, the info says it's sent successfully.

**Comment by andyko** [comment]
@tomasjkn I'll check magicdraw.log to see if I can glean anything that would help. Thanks.

**Comment by andyko** [comment]
@dlamoris You mentioned that you were fighting this 2 months ago? And it was failing on strings, reals and booleans? Are you still experiencing this? If not, do you remember what happened that fixed it?

**Comment by andyko** [comment]
Update:
Checked the log file and found an interesting message:
"Error! No element found for MMS_1423758197147_08587af3-60aa-41f1-8fd7-f025c7cc2f5e.\n[WARNING]: createOrUpdateModel deltas not posted properly\n"

So, what does it mean that "createOrUpdateModel deltas not posted properly", and how would I solve this?

Andy Ko
